tox.ini

```diff
@@ -37,3 +31,28 @@ passenv = {[testenv]passenv} BRANCH GIT_REPO DOCKER_* COMPOSE_FILE
 whitelist_externals = /bin/bash
 commands =
     bash -c "{toxinidir}/bin/docker-build.sh"
+
+[testenv:frontend-translations]
+deps =
+    nodeenv
+    {[testenv]deps}
+whitelist_externals =
+    nodeenv
+    /bin/bash
+commands =
+    nodeenv "{envtmpdir}/node_env"
+    bash -c 'source "{envtmpdir}/node_env/bin/activate" && npm --prefix "{toxinidir}/portal" install'
+    bash -c 'source "{envtmpdir}/node_env/bin/activate" && node "{toxinidir}/portal/node_modules/gulp/bin/gulp.js" --gulpfile "{toxinidir}/portal/i18next_gulpfile.js"'
```

Comment:
@ivan-c ivan, running the default gulp task here only extracts text in frontend code to a json file and then converts json file to pot file.  
Will we have another automated step that runs the conversion of po file(s) returned from Smartling into json file (containing translated text) which then can be consumed by the frontend?  Currently that is a gulp task of its own, i.e. "i18nextConvertPOToJSON".

Reply:
Correct. We will still need to manually run that task (and commit the changes) every time new translations are added to Smartling.
The problem is that we can't commit to the portal git repo from TravisCI very easily, so we'll have to do it manually for now.